src/python/pants/base/fingerprint_strategy.py

```diff
@@ -39,6 +39,8 @@ def compute_fingerprint(self, target):
   def fingerprint_target(self, target):
     """Consumers of subclass instances call this to get a fingerprint labeled with the name"""
     fingerprint = self.compute_fingerprint(target)
+    # TODO: We probably don't need to qualify the fingerprint with the strategy name any more,
```

Comment:
Yeah, qualify it with the registration name of the tool or something.

Do you agree that the problem I mention above is real? Or am I misunderstanding something about how all this works?